src/number_format.js

```diff
@@ -197,15 +198,22 @@ class NumberFormat extends React.Component {
 
   getSeparators() {
     const {decimalSeparator} = this.props;
-    let {thousandSeparator} = this.props;
+    let {thousandSeparator, allowedDecimalSeparators} = this.props;
 
     if (thousandSeparator === true) {
       thousandSeparator = ','
     }
+    if (!allowedDecimalSeparators) {
+      allowedDecimalSeparators = [decimalSeparator]
+      if (decimalSeparator !== '.') {
```

Comment:
No need for this if condition.  You can just define 
```js
allowedDecimalSeparators = ['.', decimalSeparator]
```
It's okay if dot gets duplicate as we are using this array just for checking includes.